x-pack/platform/plugins/shared/alerting/public/pages/maintenance_windows/components/maintenance_window_solution_selection.test.tsx

```diff
@@ -0,0 +1,179 @@
+        selectedSolution={'securitySolution'}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={true}
+      />
+    );
+
+    expect(screen.getByLabelText('Observability rules')).not.toBeChecked();
+    expect(screen.getByLabelText('Security rules')).toBeChecked();
+    expect(screen.getByLabelText('Stack rules')).not.toBeChecked();
+  });
+
+  it('can choose solution', async () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        selectedSolution={'observability'}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={true}
+      />
+    );
+
+    fireEvent.click(screen.getByLabelText('Stack rules'));
+
+    expect(mockOnChange).toHaveBeenLastCalledWith('management');
+
+    fireEvent.click(screen.getByLabelText('Security rules'));
+
+    expect(mockOnChange).toHaveBeenLastCalledWith('securitySolution');
+  });
+
+  it('should display loading spinner if isLoading is true', () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        isLoading
+        selectedSolution={''}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={true}
+      />
+    );
+    expect(screen.getByTestId('maintenanceWindowSolutionSelectionLoading')).toBeInTheDocument();
+  });
+
+  it('should display error message if it exists', () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        selectedSolution={''}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        errors={['test error']}
+        onChange={mockOnChange}
+        isScopedQueryEnabled={true}
+      />
+    );
+    expect(screen.getByText('test error')).toBeInTheDocument();
+  });
+
+  it('should set only 1 category at a time if scoped query is enabled', () => {
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        isScopedQueryEnabled={true}
+        selectedSolution={''}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+      />
+    );
+
+    let managementCheckbox = screen.getByLabelText('Stack rules');
+
+    fireEvent.click(managementCheckbox);
+
+    expect(mockOnChange).toHaveBeenLastCalledWith('management');
+
+    appMockRenderer.render(
+      <MaintenanceWindowSolutionSelection
+        isScopedQueryEnabled={true}
+        selectedSolution={'observability'}
+        availableSolutions={['observability', 'management', 'securitySolution']}
+        onChange={mockOnChange}
+      />
+    );
+
+    managementCheckbox = screen.getByLabelText('Stack rules');
+
+    fireEvent.click(managementCheckbox);
+
+    expect(mockOnChange).toHaveBeenLastCalledWith('management');
+  });
```

Comment:
I don't understand this test. Why do we call `render` twice? Shouldn't these be two separate unit tests?

Reply:
I've renamed the file so all old tests are changed as well. I'll think what it can mean.

Reply:
I do not understand what was intended here. But base on name I assume we cover this case but in `x-pack/platform/plugins/shared/alerting/public/pages/maintenance_windows/components/create_maintenance_windows_form.test.tsx` test `can select one in the time solution id`. So I deleted it.